deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/DeployJanitor.java

```diff
@@ -62,8 +69,12 @@ void processDeploys() throws Exception {
                         deployDAO.deleteUnusedDeploys(envId, timeThreshold, numToDelete);
                         LOG.info(String.format("Successfully removed deploys: %s before %d milliseconds has %d.",
                             envId, timeThreshold, numToDelete));
+
+                        errorBudgetSuccess.increment();
```

Comment:
seems like a double counting

Reply:
For this class, we have 2 Success and 3 Failures:

Success:
Line 73: Reports success for each item on the loop, and only if there is something to be processed. This means that attended the conditions: numToDelete > 0 AND connection != null

Line 120: Reports success if there was no error on the service, even if there was nothing to be processed by the for loop

Failure:
Line 77: Reports failure if an item on the loop raises an error
Line 106: Reports failure on the initial execution of the service, and stops the execution
Line 124: Reports failure any other execution goes wrong

Reply:
After deploying the metrics, I hope to write a quick doc regarding the rationale. Ticket: [SRE-13669](https://jira.pinadmin.com/browse/SRE-13669)

As added on the ticket comment:

- Each worker should at least report one success or failure
- We should avoid refactoring or big changes
- Each worker needs to process many items. For each item, when possible, we should report success/failure.
- If no item needs to be processed, we should report success
- We should avoid reporting a Failure and Success for the same execution